lib/ex_doc/language/elixir.ex

```diff
@@ -444,15 +444,29 @@ defmodule ExDoc.Language.Elixir do
     case Keyword.fetch(attrs, :href) do
       {:ok, href} ->
         case Regex.scan(@ref_regex, href) do
-          [[_, custom_link]] -> url(custom_link, :custom_link, config)
-          [] -> build_extra_link(href, config)
+          [[_, custom_link]] ->
+            custom_link
+            |> url(:custom_link, config)
+            |> maybe_remove_invalid_ref(href)
+
+          [] ->
+            build_extra_link(href, config)
         end
 
       _ ->
         nil
     end
   end
 
+  defp maybe_remove_invalid_ref(result, href) do
+    if is_nil(result) and String.match?(href, ~r/`.*`/) do
+      IO.warn("found invalid reference: #{href}")
```

Comment:
Could you look at the other reference warnings code and include file:line in this one too? The line isn’t perfect for these because it corresponds to where doc block starts, not where the warning happens inside the block, but that’s ok for now.